Update Gatsby from 2.32 to 4.10

ericherman opened the pull request:
After reviewing dependabot alerts, a Gatsby upgrade was called for.

After upgrading Gatsby and dependencies including commonground/design-system, some visual differences exist.

Most differences are minor, but one difference creates a usability issue.
The font size of the main navigation bar at the top of the pages is not being set correctly.
This introduces a horizontal scrollbar, and the language switching between NL and EN is scrolled off to the right.

before:
![image](https://user-images.githubusercontent.com/1174254/160583107-3d4eed1c-1b18-4def-bad3-f2baee9082eb.png)

after:
![image](https://user-images.githubusercontent.com/1174254/160583288-eefc5b53-2cb4-446c-87b2-5239b5470a3b.png)

after with scroll to right:
![image](https://user-images.githubusercontent.com/1174254/160583406-09043867-373a-4a3c-aa30-b1160fe7938d.png)

Other than font sizes, the site seems to render okay.

ericherman committed:
Update npm lockfileVersion 1 to 2

ericherman committed:
update gatsby 2.32.6 -> 4.10.3

After reviewing Dependabot alerts[0], an update to newer Gatsby was
called for, and this requires a number of Gatsby related packages as
well.

I was unable to find a set of dependencies which would work together
with Gatsby 3. However, the latest release[1] of Gatsby is 4.10 and I
was able to get packages to work for this version.

Testing with this revealed a display issue. The font size of the main
navigation bar at the top of the pages is not being set correctly.

This introduces a horizontal scrollbar, and the language switching
between NL and EN is scrolled off to the right.

Other than the display issue, the rest of the site seems to render well
enough, even though fonts in various places have different sizes.

One of the packages which is upgraded is @commonground/design-system;
this may be at the heart of the differences in display.  This upgrade
seems like quite a jump (from 12.5 to 21.1) and while an intermediate
version may work, a better approach would be to figure out how to get
the display elements to be correct with the latest version.

[0] https://github.com/Signalen/signalen.org/security/dependabot
[1] https://github.com/gatsbyjs/gatsby/releases

Ainali commented:
Hmm, while the fonts is larger, when I'm building your branch I don't get the scroll bar (instead I see all the text).
![image](https://user-images.githubusercontent.com/5566159/160587575-39a7671a-b927-4103-8d77-c58d6b3dc210.png)

ericherman committed:
add deepmerge explicitly to avoid old versions

ericherman committed:
Add optional dependency for fsevents

Not seen locally, but rather with GitHub workflow fails with:

```
npm ERR! fsevents not accessible from chokidar
```

Seems to be a platform difference.

ericherman committed:
CI workflow: update Node from 12 -> 16

ericherman committed:
Specify node version 16 for Netlify

ericherman committed:
Create a .node-version for Netlify
